pkg/controller/podgc/gc_controller.go

```diff
@@ -173,7 +175,7 @@ func (gcc *PodGCController) gcTerminating(ctx context.Context, pods []*v1.Pod) {
 		wait.Add(1)
 		go func(namespace string, name string) {
 			defer wait.Done()
-			if err := gcc.deletePod(ctx, namespace, name); err != nil {
+			if err := gcc.failAndDeletePodWithCondition(ctx, namespace, name, nil); err != nil {
```

Comment:
worth adding a `failAndDeletePod` that doesn't require a condition and calls the other function.

Reply:
updated